.env

```diff
@@ -1,4 +1,4 @@
-COMPOSE_PROJECT_NAME=sentry_onpremise
+COMPOSE_PROJECT_NAME=sentry_self_hosted
```

Comment:
Will this trip people up if they are upgrading an existing installation?

Reply:
yes, this is likely a breaking change.

Reply:
In that case it seems as if we should keep this as `sentry_onpremise` for now, put a note in the changelog for 21.12.0 and possibly a warning inline on `install.sh`, and then make the breaking change in 22.1.0.

Reply:
It would be better to output this on `install.sh`, users may probably not read the changelog on every release.

Reply:
One mention in our codebase:

https://github.com/getsentry/self-hosted/blob/459323354ef8f370828636cba29e36e90613525f/install/wrap-up.sh#L9-L10

... and then:

https://docs.docker.com/compose/reference/envvars/#compose_project_name

Reply:
Okay so how do I test this? I want to launch self-hosted using the old value, then update to the new value, then relaunch self-hosted.

Reply:
Why do we use `--network="${COMPOSE_PROJECT_NAME}_default"` in `wrap-up.sh`? 🤔 

> Your app’s network is given a name based on the “project name”, which is based on the name of the directory it lives in. You can override the project name with either the `--project-name` flag or the `COMPOSE_PROJECT_NAME` environment variable.

https://docs.docker.com/compose/networking/

👍

```
$ docker network ls
$ docker network ls
NETWORK ID     NAME                       DRIVER    SCOPE
36ae04d3f89c   bridge                     bridge    local
b14723786ff7   host                       host      local
71053c5e70ae   none                       null      local
b5034c179e6c   sentry_onpremise_default   bridge    local
$
```

Reply:
Afaict containers, networks, and volumes all use project name. 🤔

Reply:
[No volume rename](https://github.com/moby/moby/issues/31154), [best option is manual](https://serverfault.com/a/1022691):

```
docker volume create --name <new_volume>
docker run --rm -it -v <old_volume>:/from:ro -v <new_volume>:/to alpine \
    ash -c "cd /from ; cp -av . /to"
docker volume rm <old_volume>
```

Reply:
Proposal:

1. Containers—Don't care. They end up with orphans that they can clean up on their own time.
2. Network—Don't care. Spin down old, spin up new. They can clean up if they want, same as (1).
3. Volumes—This is the tricky one. The user needs to take some action in order for this to work, either they need to rename their volumes manually as just described, or they need to override `COMPOSE_PROJECT_NAME` in `.env.custom` to retain the old `sentry_onpremise` value. It sorta feels to me like adapting to config changes like this are a normal part of the upgrade process, so we are within bounds to put this burden on users. Thoughts?

Reply:
How about not changing current `onpremise` volumes and create `self-hosted` in case there aren't already volumes created?

Reply:
Good news! We create most of our volumes by [calling `docker volume create` ourselves](https://github.com/getsentry/self-hosted/blob/459323354ef8f370828636cba29e36e90613525f/install/create-docker-volumes.sh), marking them [`external`](https://docs.docker.com/compose/compose-file/compose-file-v3/#external) [in `docker-compose.yml`](https://github.com/getsentry/self-hosted/blob/459323354ef8f370828636cba29e36e90613525f/docker-compose.yml#L360-L373):

> If set to `true`, specifies that this volume has been created outside of Compose. `docker-compose up` does not attempt to create it, and raises an error if it doesn’t exist.

It [seems](https://forums.docker.com/t/docker-compose-prepends-directory-name-to-named-volumes/32835/2) that `docker-compose` prepends the project name when creating a volume, but `docker volume create` does not. 💃 

```
$ docker volume list
DRIVER    VOLUME NAME
$ ./install.sh --no-user-prompt
[...]
$ docker volume list
DRIVER    VOLUME NAME
local     f1d00624e9452e7075f7224f8eeb08744005c64938d6d2d92812d463926429b8
local     sentry-clickhouse
local     sentry-data
local     sentry-kafka
local     sentry-postgres
local     sentry-redis
local     sentry-symbolicator
local     sentry-zookeeper
local     sentry_onpremise_sentry-clickhouse-log
local     sentry_onpremise_sentry-kafka-log
local     sentry_onpremise_sentry-secrets
local     sentry_onpremise_sentry-smtp
local     sentry_onpremise_sentry-smtp-log
local     sentry_onpremise_sentry-zookeeper-log
$
```

Sooooooooo ... can we get away with orphaning the log/secrets/smtp volumes? 🧐

Reply:
The external volumes are ancient: 5d064c2224aa73a45ee0e29063be85c8810f4707.
The non-external are also ancient: #291.

Reply:
I read #291 as saying that these are ephemeral volumes that we can recreate on down/up (the reason for #291 was to avoid littering with random volumes, not to address an operational issue with non-persistent volumes).

I've confirmed that the prefixed volumes are ephemeral with this method:

1. `install.sh` on `master`
2. `dcm up`
3. [Populate](https://github.com/getsentry/self-hosted/issues/1182#issue-1068600193) a couple events.
    1.  `export SENTRY_DSN="foo"`
    2. `sentry-cli send-event -m "[Random-Error] $(echo $RANDOM | md5sum | head -c 20; echo;)" > /dev/null 2>&1 &`
4. `dcm down`
5. switch to this branch
6. `install.sh`
7. `dcm up`
8. Did events persist? Yes.
9. Populate a few more events. Did it work? Yes.

Reply:
Okay! I'm ready to say that this change is non-breaking. I've pushed 2573e87fa4738dc7ef50423c67333d14dbb1f073 but otherwise I am ready to go with this PR. Last chance to stop me @BYK @aminvakil et al. :)